[bug] sim tests with empty validator set after InitChain

## Context

Ref: https://github.com/NibiruChain/nibiru/actions/runs/16539563763/job/46778845657#logs

--- FAIL: TestAppStateDeterminism (0.49s)
panic: validator set is empty after InitGenesis, please ensure at least one validator is initialized with a delegation greater than or equal to the DefaultPowerReduction ({824640280928}) [recovered]
	panic: validator set is empty after InitGenesis, please ensure at least one validator is initialized with a delegation greater than or equal to the DefaultPowerReduction ({824640280928})

goroutine 12 [running]:
testing.tRunner.func1.2({0x260d920, 0xc001783960})
	/opt/hostedtoolcache/go/1.24.5/x64/src/testing/testing.go:1734 +0x21c
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.24.5/x64/src/testing/testing.go:1737 +0x35e
panic({0x260d920?, 0xc001783960?})
	/opt/hostedtoolcache/go/1.24.5/x64/src/runtime/panic.go:792 +0x132
github.com/cosmos/cosmos-sdk/types/module.(*Manager).InitGenesis(_, {{0x36a0fe8, 0x50406c0}, {0x36ba500, 0xc001487100}, {{0x0, 0x0}, {0x2b387af, 0xe}, 0x0, ...}, ...}, ...)
	/home/runner/go/pkg/mod/github.com/!nibiru!chain/cosmos-sdk@v0.47.11-nibiru.3/types/module/module.go:372 +0x454
github.com/cosmos/cosmos-sdk/runtime.(*App).InitChainer(_, {{0x36a0fe8, 0x50406c0}, {0x36ba500, 0xc001487100}, {{0x0, 0x0}, {0x2b387af, 0xe}, 0x0, ...}, ...}, ...)
	/home/runner/go/pkg/mod/github.com/!nibiru!chain/cosmos-sdk@v0.47.11-nibiru.3/runtime/app.go:153 +0xfe
github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).InitChain(0xc000dd21e0, {{0x0, 0x103b00a0a4, 0x5009bc0}, {0x2b387af, 0xe}, 0xc001a49540, {0x0, 0x0, 0x0}, ...})
	/home/runner/go/pkg/mod/github.com/!nibiru!chain/cosmos-sdk@v0.47.11-nibiru.3/baseapp/abci.go:78 +0x4f5
github.com/cosmos/cosmos-sdk/x/simulation.initChain(0xc0011aed20, {0x3fec3394d78d0149, _, _, {_, _, _}, {_, _}, {_, ...}}, ...)
	/home/runner/go/pkg/mod/github.com/!nibiru!chain/cosmos-sdk@v0.47.11-nibiru.3/x/simulation/simulate.go:46 +0x285
github.com/cosmos/cosmos-sdk/x/simulation.SimulateFromSeed({0x36cbf40, 0xc00107afc0}, {_, _}, _, _, _, {0xc0015b2308, 0x2d, 0x2f}, ...)
	/home/runner/go/pkg/mod/github.com/!nibiru!chain/cosmos-sdk@v0.47.11-nibiru.3/x/simulation/simulate.go:80 +0x4af
github.com/NibiruChain/nibiru/v2/app/simapp_test.TestAppStateDeterminism(0xc00107afc0)
	/home/runner/work/nibiru/nibiru/app/simapp/sim_test.go:152 +0x885
testing.tRunner(0xc00107afc0, 0x3276428)
	/opt/hostedtoolcache/go/1.24.5/x64/src/testing/testing.go:1792 +0xf4
created by testing.(*T).Run in goroutine 1
	/opt/hostedtoolcache/go/1.24.5/x64/src/testing/testing.go:1851 +0x413
FAIL	github.com/NibiruChain/nibiru/v2/app/simapp	0.594s
FAIL
make: *** [contrib/make/simulation.mk:6: test-sim-nondeterminism] Error 1

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -52,6 +52,7 @@ See https://github.com/dangoslen/changelog-enforcer.
 - [#2344](https://gittub.com/NibiruChain/nibiru/pull/23344) - feat(evm): Add some evm messages into the evm codec.
 - [#2346](https://gittub.com/NibiruChain/nibiru/pull/2346) - fix(buf-gen-rs): improve Rust proto binding generation script robustness and get it to work with a forked Cosmos-SDK dependency and exit correctly on failure
 - [#2348](https://github.com/NibiruChain/nibiru/pull/2348) - fix(oracle): max expiration a label rather than an invalidation for additional query liveness
+- [#2350](https://github.com/NibiruChain/nibiru/pull/2350) - fix(simapp): sim tests with empty validator set panic
 
 ### Dependencies
 - Bump `form-data` from 4.0.1 to 4.0.4 ([#2347](https://github.com/NibiruChain/nibiru/pull/2347))

--- a/app/simapp/state_test.go
+++ b/app/simapp/state_test.go
@@ -35,6 +35,15 @@ func AppStateFn(cdc codec.JSONCodec, simManager *module.SimulationManager) simty
 			genesisTimestamp = time.Unix(appsim.FlagGenesisTimeValue, 0)
 		}
 
+		if len(accs) == 0 {
+			// Fallback to generated accounts if none were passed in
+			// We do this because staking will fail if there are 0 accounts to be
+			// eligible validators, causing a panic during InitChain.
+			//
+			// See https://github.com/NibiruChain/nibiru/issues/2349
+			accs = simtypes.RandomAccounts(r, 3)
+		}
+
 		chainID = config.ChainID
 		switch {
 		case config.ParamsFile != "" && config.GenesisFile != "":
@@ -156,7 +165,11 @@ func AppStateRandomizedFn(
 	)
 	appParams.GetOrGenerate(
 		cdc, sims.InitiallyBondedValidators, &numInitiallyBonded, r,
-		func(r *rand.Rand) { numInitiallyBonded = int64(r.Intn(300)) },
+		func(r *rand.Rand) {
+			// Note that we set numInitiallyBonded to be at least 1, because a chain
+			// with 0 validators is invalid an will panic in InitChain.
+			numInitiallyBonded = max(int64(r.Intn(300)), 1)
+		},
 	)
 
 	if numInitiallyBonded > numAccs {

